### ISSUE TITLE
Setting OutputType WinExe is not preventing a console host

### ISSUE THREAD

**Issue by ruofanwang-pp** [dotnet/wpf#3944]
**.NET Core Version: (e.g. 3.0 Preview1, or daily build number, use `dotnet --info`)**
[Docker image for dotnet-sdk-LTS 3.1](mcr.microsoft.com/dotnet/sdk:3.1)

**Windows version: (`winver`)**
[Github Hosted Environment](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources) 
[Winders Server 2019 - OS Version: 10.0.17763 Build 1637](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md)

**Does the bug reproduce also in WPF for .NET Framework 4.8?:**
~Yes/No~ Did not test

**Is this bug related specifically to tooling in Visual Studio (e.g. XAML Designer, Code editing, etc...)? If yes, please file the issue via the instructions [here](https://docs.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019).** 
No, but it is working as intended when manually publishing on on my local machine (Windows 10 202h).

**Security issues and bugs should be reported privately, learn more via our [responsible disclosure guidelines](https://github.com/dotnet/wpf/blob/master/README.md#reporting-security-issues-and-security-bugs).**
-

 
 
 **Problem description:**
The project is using `Microsoft.NET.Sdk.WindowsDesktop`  with the `OutputType` set to `WinExe`. The produced executable from Github Workflow with Docker is opening the application with a console host. 
 
```
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <UseWPF>true</UseWPF>
    <ApplicationIcon>Common\Images\favicon.ico</ApplicationIcon>
  </PropertyGroup>

  <ItemGroup>
...
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.9" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="3.1.9" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="3.1.9" />
    <PackageReference Include="ReactiveUI.Events.WPF" Version="11.5.35" />
    <PackageReference Include="ReactiveUI.WPF" Version="11.5.35" />
    <PackageReference Include="Serilog.Extensions.Hosting" Version="3.1.0" />
    <PackageReference Include="Serilog.Sinks.Debug" Version="1.0.1" />
    <PackageReference Include="Splat.Microsoft.Extensions.DependencyInjection" Version="9.6.1" />
    <PackageReference Include="Splat.Microsoft.Extensions.Logging" Version="9.6.1" />
    <PackageReference Include="System.Linq.Async" Version="4.1.1" />
  </ItemGroup>

  <ItemGroup>
    <Resource Include="Common\Fonts\39784E_0_0.ttf" />
    <Resource Include="Common\Fonts\39784E_1_0.ttf" />
  </ItemGroup>

  <ItemGroup>
...
  </ItemGroup>

  <ItemGroup>
...  </ItemGroup>

</Project>
```

 **Actual behavior:** 
 The produced executable from our Github workflow is opening the application with a console host.

 **Expected behavior:**
 The produced executable should not open a console host.

 **Minimal repro:**
100% on 20+ runs. 
- Same actual outputs with publishing profile
- Old docker image `mcr.microsoft.com/windows/servercore:ltsc2019`

**Comment by thomasclaudiushuber** [comment]
Hi @ruofanwang-pp , sounds more like a GitHub issue than a .NET or WPF issue. Did you look already at the stuff at https://github.com/microsoft/github-actions-for-desktop-apps?
Maybe it's worth to post an issue there.

**Comment by lindexi** [comment]
@ruofanwang-pp  Could you add `<DisableWinExeOutputInference>true</DisableWinExeOutputInference>` to your csproj file? See: [Breaking change: OutputType set to WinExe for WPF and WinForms apps](https://docs.microsoft.com/en-us/dotnet/core/compatibility/windows-forms/5.0/automatically-infer-winexe-output-type?WT.mc_id=DX-MVP-5003606)

**Comment by ruofanwang-pp** [comment]
> Hi @ruofanwang-pp , sounds more like a GitHub issue than a .NET or WPF issue. Did you look already at the stuff at https://github.com/microsoft/github-actions-for-desktop-apps?
> Maybe it's worth to post an issue there.

Thanks for your reply, @thomasclaudiushuber. I'll also  try to reproduce this issue without docker and straight with Github Actions. I'll post an issue there, if the same actual behavior is reproduced.

**Comment by davidathompson** [comment]
Yes the /dotnet/sdk:5.0 is doing the same thing, I can't get a winexe to compile and not open up console window first

**Comment by lindexi** [comment]
How about set DisableWinExeOutputInference property

```xml
    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <DisableWinExeOutputInference>true</DisableWinExeOutputInference>
        <TargetFramework>net5.0-windows</TargetFramework>
        <UseWPF>true</UseWPF>
    </PropertyGroup>
```

See [Breaking change: OutputType set to WinExe for WPF and WinForms apps](https://docs.microsoft.com/en-us/dotnet/core/compatibility/windows-forms/5.0/automatically-infer-winexe-output-type?WT.mc_id=DX-MVP-5003606)

**Comment by davidathompson** [comment]
@lindexi Thanks for the idea. Tried this as well. The same dotnet publish command will work fine on a regular machine but the same command, run inside the sdk container, will always show the console window.
